File: src/main/java/com/fazula/server/common/jsonutils/JsonUtils.java
package com.fazula.server.common.jsonutils;

import io.vertx.core.json.Json;
import io.vertx.core.json.JsonArray;
import io.vertx.core.json.JsonObject;
import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.Stream;

public class JsonUtils {
    public JsonUtils() {
    }

    public static String encodeResult(Object result) {
        if (result == null) {
            return null;
        } else {
            return result instanceof List ? (new JsonArray(Json.encode(result))).encodePrettily() : Json.encode(result);
        }
    }

    public static String getAuthorizationHeader(JsonObject message) {
        return message.getString("authorization").replace("Bearer", "").trim();
    }

    public static Stream<JsonObject> iterateJsonArray(JsonArray jsonArray) {
        IntStream var10000 = IntStream.range(0, jsonArray.size());
        jsonArray.getClass();
        return var10000.mapToObj(jsonArray::getJsonObject);
    }
}

File: nyala-common/src/main/java/com/nyala/common/jsonutils/JsonUtils.kt
package com.nyala.common.jsonutils

import io.vertx.core.json.Json
import io.vertx.core.json.JsonArray
import io.vertx.core.json.JsonObject
import java.util.stream.IntStream
import java.util.stream.Stream

object JsonUtils {
    @JvmStatic
    fun encodeResult(result: Any?): String? {
        return if (result == null) {
            null
        } else {
            if (result is List<*>) JsonArray(Json.encode(result)).encodePrettily() else Json.encode(result)
        }
    }

    fun getAuthorizationHeader(message: JsonObject): String {
        return message.getString("authorization").replace("Bearer", "").trim { it <= ' ' }
    }

    fun iterateJsonArray(jsonArray: JsonArray): Stream<JsonObject> {
        val list = IntStream.range(0, jsonArray.size())
        jsonArray.javaClass
        return list.mapToObj { pos: Int -> jsonArray.getJsonObject(pos) }
    }
}